public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts

```diff
@@ -39,24 +39,19 @@ export class CompletionDataProvider {
   }
 
   async getLabelNames(otherLabels: Label[] = []) {
-    if (otherLabels.length === 0) {
-      // if there is no filtering, we have to use a special endpoint
-      return this.languageProvider.getLabelKeys();
-    }
```

Comment:
Here we want to keep `this.languageProvider.getLabelKeys()`; as it provides cached loaded/cached labels, while your implementation re-requests these labels.

Reply:
fixed, thanks